loglist2/loglist2.go

```diff
@@ -136,28 +150,38 @@ type TreeHead struct {
 	TreeSize int64 `json:"tree_size"`
 }
 
-// String method returns printable name of the state.
-func (ls *LogStates) String() string {
+// Which method returns Log-status for descriptive struct.
+func (ls *LogStates) Which() LogStatus {
 	switch {
 	case ls == nil:
-		return "Empty"
+		return UndefinedLogStatus
 	case ls.Pending != nil:
-		return "Pending"
+		return PendingLogStatus
 	case ls.Qualified != nil:
-		return "Qualified"
+		return QualifiedLogStatus
 	case ls.Usable != nil:
-		return "Usable"
+		return UsableLogStatus
 	case ls.Frozen != nil:
-		return "Frozen"
+		return FrozenLogStatus
 	case ls.Retired != nil:
-		return "Retired"
+		return RetiredLogStatus
 	case ls.Rejected != nil:
-		return "Rejected"
+		return RejectedLogStatus
 	default:
-		return "Empty"
+		return UndefinedLogStatus
 	}
 }
 
+// String returns printable name for log-state.
+func (ls LogStatus) String() string {
```

Comment:
Group this up with the definition of `LogStatus`? 

Also, this looks like the *perfect* use case for https://godoc.org/golang.org/x/tools/cmd/stringer

Reply:
Removed. As stringer puts its production in a separate file.